src/Cms/LanguageVariable.php

```diff
@@ -102,10 +102,10 @@ public function key(): string
 	/**
 	 * Sets a new value for the language variable
 	 */
-	public function update(string $value): static
+	public function update(string|null $value = null): static
 	{
-		$translations = $this->language->translations();
-		$translations[$this->key] = $value;
+		$translations             = $this->language->translations();
+		$translations[$this->key] = $value ?? '';
```

Comment:
`Question:` Do we need the fallback to an empty string? What happens when it just stays null?

`Todo:` We would need to add a unit test for passing null to `::update()`

Reply:
If stays `null`, language variables comes `null`. Do you think it's ok? I don't think so.

````php
'translations' => [
    'test-variable' => NULL
],
````

Reply:
So how does that array look before when you got the error? I'd think it's also `=> NULL` then that let to the error, isn't it?

Reply:
No. When adding a variable (without a value) for the default language, an empty string is already added:
https://github.com/getkirby/kirby/blob/fix/updating-null-language-variables/config/areas/languages/dialogs.php#L195

However, since that variable does not exist in the secondary language yet, it returns `null` and if no update is made in the dialog, it tries to save `null`.

`create` solved this in the dialog process, we can solve it the same way for `update`.

Reply:
Let me have a closer look during the week. I think we have a problem if the dialogs behave differently than calling the methods programmatically. That's why I think it makes sense to spend some time here finding the ideal solution - all your solutions are functional patches for sure.

Reply:
@afbora finally had a closer look: I agree with you

- `language.translation.update` dialog: in the `submit` handler, we should probably add an empty string as fallback `->get('value', '')` to match the behavior of the create dialoh
- `LanguageVariable::update()` should indeed add a fallback to an empty string as you did.
- Should it also add `trim()` for the value as `::create()` does?

Reply:
Actually we should remove the `trim` function for create method. I think user is responsible for the spaces. May be user needs spaces start/end of string. What do you think? @distantnative

Reply:
True, could be a valid use case. So aas you said, maybe remove it from create.

Reply:
What do you think? @bastianallgeier